src/runtime/components/AlgoliaDocSearch.vue

```diff
@@ -7,39 +7,113 @@
 <script setup lang="ts">
 import type { PropType } from 'vue'
 import { withoutTrailingSlash } from 'ufo'
-import type { DocSearchOptions } from '../../types'
+import { DocSearchProps, type docsearch as docsearchFunc } from 'docsearch'
+import type { DocSearchTranslations } from '@docsearch/react'
+import type { HitComponentFunc, ModuleBaseOptions, SearchOptions } from '../../types'
 // @ts-ignore - These are Nuxt3 aliases
-import { useRuntimeConfig, useRoute, useRouter, onMounted, watch, computed } from '#imports'
+import { useRuntimeConfig, useRoute, useRouter, onMounted, watch } from '#imports'
 
 const route = useRoute()
-
 const router = useRouter()
 
 const props = defineProps({
-  options: {
-    type: [Object, undefined] as PropType<DocSearchOptions | undefined>,
-    required: false,
+  applicationId: {
+    type: String,
+    default: () => (useRuntimeConfig().public.algolia as ModuleBaseOptions)?.applicationId
```

Comment:
I know that `(useRuntimeConfig().public.algolia as ModuleBaseOptions)` is very repetitive, but I couldn't refactor this into a composable in this file, because the vue compiler gave a warning about function hoisting. 

The only option seems to me to create a composable (call for example `useDocSearchConfig`) in another file, and import it, but I didn't want to introduce a different kind of composable which we don't even want to expose to the users.